### DISCUSSION TITLE
I installed the deb version of Freeplane. Can it automatically upgrade itself on my Linux Mint computer?

### DISCUSSION THREAD

**Discussion by hub2git** [freeplane/freeplane#2119] [Help]
I'm on Linux Mint 22 Cinnamon.
Synaptic package manager has 1.7.X of Freeplane. I installed it, then when I was reading the Freeplane changelog, i realized that the newest version is way newer (currently 1.12.6). So I uninstalled the 1.7.X freeplane from Synaptic and downloaded https://cfhcable.dl.sourceforge.net/project/freeplane/freeplane%20stable/freeplane_1.12.6~upstream-1_all.deb?viasf=1

Happily, it installed and worked fine without  my need to manually do additional steps (like messing with environmental path or installing Java).

Can I get Freeplane to autoupdate? I have an Android phone and I like how it will autoupgrade my apps. Or, on Linux Mint, programs  i get from synaptic will auto-upgrade.

**Comment by macmarrum** [comment]
The Freeplane package in Debian (and its derivatives, like Linux Mint) is outdated and until someone volunteers to keep maintaining it, it will likely stay that way → https://github.com/freeplane/freeplane/discussions/275#discussioncomment-3032928

I'm not aware of any apt-compatible repositories with the .deb packages offered on https://sourceforge.net/projects/freeplane/files/freeplane%20stable/. If such a repository existed, it could be added as a source to Synaptic, and automatic updates would work.

But Freeplane will indicate (in the status bar) when a new version is available and you could use a script to download it from sourceforge.net and then install it manually.

Here's a _downloadFreeplane.groovy_ script that I've been using. You can run it in Freeplane. Alternatively, outside of Freeplane, via Groovy.

```groovy
// @ExecutionModes({ON_SINGLE_NODE})
/* downloadFreeplane.groovy
 * Copyright (C) 2024  macmarrum (at) outlook (dot) ie
 * SPDX-License-Identifier: GPL-3.0-or-later
 *
 * Set the following options according to your needs (lines 35 - 38)
 * - filesPageUrl
 * - artifactToDownload
 * - downloadsDir
 */

// https://wjw465150.github.io/blog/Groovy/my_data/Robust_HTML_parsing_the_Groovy_way.htm
// https://stackoverflow.com/questions/68340139/how-to-add-ivy-as-module-dependency-to-intellij

import groovy.xml.XmlSlurper
import groovy.xml.slurpersupport.GPathResult

@Grab('org.ccil.cowan.tagsoup:tagsoup:1.2.1')

def sfFilesPage = [
        STABLE : 'https://sourceforge.net/projects/freeplane/files/freeplane%20stable/',
        PREVIEW: 'https://sourceforge.net/projects/freeplane/files/freeplane%20preview/',
]

def artifactType = [
        BIN            : /freeplane_bin/,
        DEB            : /~upstream-1_all\.deb/,
        PORTABLE       : /FreeplanePortable-/,
        SETUP_WITH_JAVA: /Freeplane-Setup-With-Java-/,
        SETUP          : /Freeplane-Setup-\d/,
        DMG_APPLE      : /(?<!JDK\d\d)-apple\.dmg/,
        DMG_INTEL      : /(?<!JDK\d\d)-intel\.dmg/,
]

//def filesPageUrl = sfFilesPage.PREVIEW
def filesPageUrl = sfFilesPage.STABLE
def artifactToDownload = artifactType.DEB
def downloadsDir = System.env.HOME + '/Downloads'

def tagsoupParser = new org.ccil.cowan.tagsoup.Parser()
GPathResult html = new XmlSlurper(tagsoupParser).parse(filesPageUrl)
GPathResult file_list = html.depthFirst().find { it.@id = 'files_list' }
GPathResult artifact_tr = file_list.depthFirst().find { it.name() == 'tr' && (it.@title as String) =~ artifactToDownload }
if (artifact_tr === null)
    throw new RuntimeException("artifact $artifactToDownload not found on $filesPageUrl")
String downloadUrl = artifact_tr.th.a.@href
def target = new File(downloadsDir, artifact_tr.@title as String)
if (target.exists()) {
    println "$target already exists"
} else {
    println "downloading to $target ... "
    target.withDataOutputStream { dos ->
        downloadUrl.toURL().eachByte(16 * 1024) { data, count ->
            dos.write(data, 0, count)
        }
    }
    println 'done'
}
```
